### PR TITLE
Codacy refactor

### LINKED ISSUE #24: Deal with Codacy
Codacy bad

Come up with a good config that doesn't cry about everything. Also DON'T USE READONLY ON STRUCTS IT BREAKS THINGS.

### PR DESCRIPTION
**Pull request by liamdalg**
Deals with most of the codacy issues (where sensible) - lots of files were changed so probably needs reviewing/testing.

Take note of the `Axle` class in `DriveController`:

```cs
[Serializable]
public class Axle
{
    [field: SerializeField] public WheelCollider LeftWheel { get; private set; }
    ...
}
```

The `[field: SerializeField]` allows for serialization on C# auto-properties, although IntelliJ doesn't like it so just ignore the complaints. Properties are good so we should use them instead of `public` fields imo.

Resolves #24

### CODE REVIEW

**Review by RileyEv — APPROVED** [review]
Codacy is poo

File: ProjectFiles/Assets/Scripts/Car/DriveController.cs [hunk 1]
```diff
@@ -1,80 +1,84 @@
...
-        else if (v > 40)
-            return 5;
-        else if (v > 80)
-            return 7;
-        else if (v > 200)
-            return 10;
-        else if (v > 400)
-            return 20;
-        else
+        private float TurnMultiplier(float v)
```
Comment (ChrisGora) [inline ProjectFiles/Assets/Scripts/Car/DriveController.cs:49]: Ewwww

File: ProjectFiles/Assets/Scripts/Car/SelfRighting.cs [hunk 2]
```diff
@@ -1,33 +1,32 @@
...
+            rb = car.GetComponent<Rigidbody>();
+            carCollider = car.GetComponent<Collider>();
+        }
+    
+        private void Update()
+        {
+            if(IsGrounded())
+            {
+                // ??
+            }
```
Comment (ChrisGora) [inline ProjectFiles/Assets/Scripts/Car/SelfRighting.cs:24]: This should be commented out or deleted

**Review by ChrisGora — CHANGES_REQUESTED** [review]
I like this refactor :heart:

**Pull request merged**